crates/cheatcodes/src/inspector.rs

```diff
@@ -753,6 +791,12 @@ impl Cheatcodes {
             }
         }
 
+        // Store the total gas used for all active gas records started by `startSnapshotGas`
+        self.gas_metering.gas_records.iter_mut().for_each(|record| {
+            record.gas_used = record.gas_used.saturating_add(outcome.result.gas.spent());
+            record.depth = ecx.journaled_state.depth();
```

Comment:
why do we change the depth here? we should insted use `record.depth` to determine whether the call gas needs to be recorded

probably something like `if record.depth + 1 == ecx.journaled_state.depth()`